docs/cli/code-push/promote.md

```diff
@@ -71,27 +57,46 @@ If no `targetDescriptors` nor a `targetSemVerDescriptor` is specified, the comma
 
-`--label/-l`
+`--sourceDeploymentName`
 
-* Promote the release matching this specific label.
-* **Default** The latest release matching sourceDescriptor/sourceDeploymentName pair will be promoted.
+* The deployment name of the release to promote (Staging for example).
+* **Default** The command will prompt to input the deployment name, or display a list of deployment names stored in the Cauldron, to choose from.
 
-`--description/--des`
+`--sourceDescriptor <descriptor>`
 
-* Description of the changes made to the app with this release. If omitted, the description from the release being promoted will be used.
+* Specify the native application version from which to promote a release in the form of a *complete native application descriptor*.
+* The release to be promoted will be the latest non disabled release of this native application version.
+* **Default** The command will list all released native applications versions stored in the Cauldron and will prompt to select a target native application version from the list.
 
-* **Default** Empty string
+`--targetBinaryVersion/-t <targetBinaryVersion>`
 
-`--disableDuplicateReleaseError`
+* Semver expression that specifies the binary app version this promotion is targeting
+* If omitted, and `reuseReleaseBinaryVersion` is not set, the promotion will target the exact version of the descriptor
+* If `versionModifier` is specified in the codePush config, it will be applied
+* For using `targetBinaryVersion` option users must target only 1 descriptor
+* For using `targetBinaryVersion` option users cannot use semVerDescriptor
+* This option is mutually exclusive with `reuseReleaseBinaryVersion`
 
-* When this flag is set, promoting a package that is identical to the latest release on the target deployment will produce a warning instead of an error
+`--targetDeploymentName`
 
-* **Default** false
+* The deployment name to promote the release to (Production for example).
+* **Default** The command will prompt to input the deployment name, or display a list of deployment names stored in the Cauldron, to choose from.
+
+`--targetDescriptors <descriptors..>`
+
+* Specify one or more target native application version to promote the release to, in the form of a *complete native application descriptor* list (separated by spaces).
+* The target descriptor can be the same as the source descriptor if the promotion is only changing the deployment name (for example promoting a release from Staging to Production for the same native application version).
+
+`--targetSemVerDescriptor <descriptor>`
+
+* A native descriptor using a semantic version string for its version. The promotion will target all native application versions matching the semver.
+
+* If no `targetDescriptors` nor a `targetSemVerDescriptor` is specified, the command will list all released native application versions stored in the Cauldron and will display a prompt to select one or more target native application version(s) for the promotion.
```

Comment:
Just a minor request: It looks like a rather big _reordering_ of existing content **and** the _addition_ of new content happened in the **same** commit. To make it easier to review, next time please use two separate commits. No need to change here :)

Reply:
Yeah, different commits would have been relevant here.